work_queue/src/json_server.py

```diff
@@ -0,0 +1,86 @@
+import socket
+import json
+
+class WorkQueueServer:
+
+    def __init__(self):
+        self.socket = socket.socket()
+        self.id = 1
+
+    def connect(self, address, port):
+        self.socket.connect((address, port))
+
+    def submit(self, task):
+        request = {
+            "jsonrpc" : "2.0",
+            "method" : "submit",
+            "id" : self.id,
+            "params" : task
+        }
+
+        request = json.dumps(request)
+
+        request += "\n"
+        self.socket.send(request)
```

Comment:
There is a lurking problem here: the TCP connection is a continuous stream, so there is no guarantee that when the sender writes N bytes, the receiver will read back exactly N bytes.  It probably works for small test cases on a single machine, but sooner or later, you will send() 100 bytes, and the recv() will only give back the first 50, and then chaos results.

What you need to do here is construct a small "envelope" that describes the length of the message, and then sends the contents of the message.  Then, when receiving, read back the fixed integer, and then read until you get exactly that number of bytes.  Then you know that you have a full message.